dev/generate_bom.py

```diff
@@ -49,17 +64,68 @@ class BomGenerator(Annotator):
 
   def __init__(self, options):
     self.__base_dir = options.base_dir
+    self.__gcr_project = options.gcr_project
+    self.__docker_registry = options.docker_registry
     self.__bom_file = ''
     self.__component_versions = {}
     super(BomGenerator, self).__init__(options)
 
   @classmethod
   def init_argument_parser(cls, parser):
     """Initialize command-line arguments."""
-    parser.add_argument('--base_dir', default='.',
+    parser.add_argument('--base_dir', default='.', required=True,
                         help="Base directory containing the component's git repositories as subdirectories.")
+    parser.add_argument('--gcr_project', default='',
+                        help="GCR project we use in Google Container Builder configuration files.")
```

Comment:
I think another (maybe more future-proof) way of doing this would be to have `--docker_registry` be required for any docker build (in gcr world it would be `gcr.io/<project>`, in docker it would be `index.docker.io/<user>`, etc...), and then have a second flag indicating which builder to use. Right now, that second flag would only support `gcr` and `local`, but could be expanded to work for quay, dockerhub, etc...